src/App.tsx

```diff
@@ -21,24 +22,42 @@ i18n.use(initReactI18next).init({
 });
 
 const App: React.FC = () => {
-  const [url, setUrl] = useState<string>('');
+  // Initialize URL from query parameter at component creation
+  // Using lazy initialization to avoid calling getInitialUrl on every render
+  const [url, setUrl] = useState<string>(() => {
+    const params = new URLSearchParams(window.location.search);
+    const urlParam = params.get('url');
```

Comment:
Is it possible to use an enum or something alike to avoid the string parameter which could be anything? Or is type and name safety guaranteed here?